ui/src/App.js

```diff
@@ -30,37 +30,62 @@ class App extends Component {
     }
   }
 
-  connect(port) {
-    window.serial.requestPort().then(selectedPort => {
-      var port = selectedPort;
-      console.log('Connecting to ' + port.device_.productName + '...');
-      port.connect().then(() => {
-        console.log('Connected to port:', port);
-        port.onReceive = this.handleSerialMessage.bind(this)
-        port.onReceiveError = this.handleSerialError.bind(this)
-
-        // Try to load our bitcoin address
-        let textEncoder = new TextEncoder();
-        let payload = textEncoder.encode("addr")
-        port.send(payload)
-            .catch(error => console.log("Error requesting Bitcoin address", error))
-
-      }, error => {
-        console.log('Connection error: ' + error);
-      });
+  componentDidMount() {
+    // Attempt to reconnect to known devices
+    this.reconnect()
+    // Watch for usb disconnections and handle them
+    navigator.usb.ondisconnect = this.handleDisconnect.bind(this)
```

Comment:
`componentDidMount` is basically react's page load event.

Eventually we should have all `navigator.usb` magic off in its own class, but for now this works.